CMakeLists.txt

```diff
@@ -58,6 +58,9 @@ message(STATUS "RMM: RMM_LOGGING_LEVEL = '${RMM_LOGGING_LEVEL}'")
 # cudart can be linked statically or dynamically
 option(CUDA_STATIC_RUNTIME "Statically link the CUDA runtime" OFF)
 
+option(RMM_SUPPORTS_LOGGING "Enable logging support" ON)
+option(RMM_SPDLOG_DYNAMIC "Use dynamic linking for spdlog" OFF)
```

Comment:
This allows local builds to choose to use a dynamic spdlog and also use one that already exists on the system. If this is off, we currently always download for static linking under the assumption that we prioritize symbol hiding.